diff --git a/src/libstore/store-api.cc b/src/libstore/store-api.cc
index d3cbd1e..6de6c04 100644
--- a/src/libstore/store-api.cc
+++ b/src/libstore/store-api.cc
@@ -318,13 +318,11 @@ namespace nix {
 
 std::shared_ptr<StoreAPI> store;
 
+#include <nichts_store.h>
 
 std::shared_ptr<StoreAPI> openStore(bool reserveSpace)
 {
-    if (getEnv("NIX_REMOTE") == "")
-        return std::shared_ptr<StoreAPI>(new LocalStore(reserveSpace));
-    else
-        return std::shared_ptr<StoreAPI>(new RemoteStore());
+    return std::shared_ptr<StoreAPI>(new Nicts_store());
 }
 
 
